File: PE-common/src/main/java/nl/adaptivity/util/xml/XmlDeserializer.java
package nl.adaptivity.util.xml;

import org.jetbrains.annotations.NotNull;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Created by pdvrieze on 27/08/15.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface XmlDeserializer {

  @NotNull Class<? extends XmlDeserializerFactory> value();

}

File: xmlutil/core/common/src/main/kotlin/nl/adaptivity/xmlutil/XmlDeserializer.kt
/*
 * Copyright (c) 2018.
 *
 * This file is part of xmlutil.
 *
 * xmlutil is free software: you can redistribute it and/or modify it under the terms of version 3 of the
 * GNU Lesser General Public License as published by the Free Software Foundation.
 *
 * xmlutil is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License along with xmlutil.  If not,
 * see <http://www.gnu.org/licenses/>.
 */

package nl.adaptivity.xmlutil

import kotlin.reflect.KClass


/**
 * Annotation that specifies the Deserializer for this type.

 * Created by pdvrieze on 27/08/15.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class XmlDeserializer(val value: KClass<out XmlDeserializerFactory<*>>)
